src/machine-learning/machine-learning.ts

```diff
@@ -252,13 +295,73 @@ export class ModelOptions {
   public displayName?: string;
   public tags?: string[];
 
-  public tfliteModel?: { gcsTFLiteUri: string; };
+  public tfliteModel?: { gcsTfliteUri: string; };
+}
 
-  protected toJSON(forUpload?: boolean): object {
-    throw new Error('NotImplemented');
+async function convertOptionstoContent(
+    options: ModelOptions, forUpload?: boolean,
+    storageClient?: StorageClient): Promise<object> {
+  const modelContent = deepCopy(options);
+  if (forUpload && modelContent.tfliteModel?.gcsTfliteUri) {
+    if (!storageClient) {
+      throw new FirebaseMachineLearningError(
+        'invalid-argument',
+        'Must specify storage client if forUpload and gcs Uri are specified.',
+      );
+    }
+    modelContent.tfliteModel.gcsTfliteUri = await signUrl(modelContent.tfliteModel.gcsTfliteUri, storageClient!);
+  }
+  return modelContent;
+}
+
+async function signUrl(unsignedUrl: string, storageClient: StorageClient): Promise<string> {
+  const MINUTES = 60 * 1000; // A minute in milliseconds.
```

Comment:
Call it MINUTES_IN_MILLIS and drop the comment.

Reply:
done